packages/shared/src/hooks/useActions.ts

```diff
@@ -28,7 +28,7 @@ export const useActions = (): UseActions => {
       client.setQueryData<Action[]>(
         generateQueryKey(RequestKey.Actions, user),
         (data) => {
-          if (!data?.length) return [];
+          if (!Array.isArray(data)) return [];
```

Comment:
In that case, should the fix be as below? Yeah, I totally missed this one.
```suggestion
          if (!Array.isArray(data)) return [{ userId: user.id, type, completedAt: new Date() }];
```